holoviews/core/util.py

```diff
@@ -886,3 +886,27 @@ def arglexsort(arrays):
         recarray['f%s' % i] = array
     return recarray.argsort()
 
+
+def get_dynamic_item(map_obj, dimensions, key):
+    """
+    Looks up an item in a DynamicMap given a list of dimensions
+    and a corresponding key. The dimensions must be a subset
+    of the map_obj key dimensions.
+    """
```

Comment:
An example would be nice in the docstring here. Doesn't have to be an actual doctest if that is too tricky to get working...